src/components/cl/doca_urom/cl_doca_urom_team.c

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2020-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "cl_doca_urom.h"
+#include "utils/ucc_malloc.h"
+#include "core/ucc_team.h"
+
+UCC_CLASS_INIT_FUNC(ucc_cl_doca_urom_team_t, ucc_base_context_t *cl_context,
+                    const ucc_base_team_params_t                *params)
+{
+    union doca_data             cookie = {0};
+    doca_error_t                result = DOCA_SUCCESS;
+    ucc_cl_doca_urom_context_t *ctx    =
+        ucc_derived_of(cl_context, ucc_cl_doca_urom_context_t);
+    ucc_status_t                status;
+
+    UCC_CLASS_CALL_SUPER_INIT(ucc_cl_team_t, &ctx->super, params);
+
+    self->teams = (ucc_team_h **) ucc_malloc(
+                        sizeof(ucc_team_h *) * UCC_CL_DOCA_UROM_MAX_TEAMS);
+
+    if (!self->teams) {
+        cl_error(cl_context->lib,
+                 "failed to allocate %zd bytes for doca_urom teams",
+                 sizeof(ucc_team_h *) * UCC_CL_DOCA_UROM_MAX_TEAMS);
+        status = UCC_ERR_NO_MEMORY;
+        return status;
+    }
+
+    self->n_teams = 0;
+    self->score_map = NULL;
+
+    cookie.ptr = &self->res;
+
+    /* Send the command to create a team on the DPU */
+    result = ucc_cl_doca_urom_task_team_create(
+                ctx->urom_ctx.urom_worker,
+                cookie,
+                ctx->urom_ctx.ctx_rank,
+                0 /* start */,
+                1 /* stride */,
+                params->params.oob.n_oob_eps /* size */,
+                ctx->urom_ctx.urom_ucc_context,
+                ucc_cl_doca_urom_team_create_finished);
+
+    if (result != DOCA_SUCCESS) {
+        cl_error(cl_context->lib, "Failed to create UCC team task");
+        return UCC_ERR_NO_RESOURCE;
+    }
+
+    self->res.team_create.status = 1; // set in progress
```

Comment:
ucc_cl_doca_urom_team_create_finished is a callback function, it only gets called when the DPU worker has sent us the result of the team create command. On the other hand, ucc_cl_doca_urom_team_create_test is called in a polling fashion. If I dont set the status to in progress here, team_create_test will get an uninitialized value

Reply:
thank you for pointing this out. it turns out, team_create.status is unnecessary because the doca status is enough. i removed it

Reply:
but the reason there needs to be a status set here is because the ucc_cl_doca_urom_team_create_finished only gets called when there is a reply from the DPU. if this was uninitialized, then team_create_test would see an uninitialized value each time it's called